app/models/data_request.rb

```diff
@@ -30,13 +45,33 @@ def kase
 
   private
 
+  def validate_from_date_before_to_date
+    if date_from.present? && date_to.present? && date_from >= date_to
+      errors.add(
+        :date_from,
+        I18n.t('activerecord.errors.models.data_request.attributes.date_from.order')
+      )
+      errors[:date_from].any?
+    end
+  end
+
+  def validate_request_type_note
+    if request_type == 'other' && request_type_note.blank?
+      errors.add(
+        :request_type_note,
+        I18n.t('activerecord.errors.models.data_request.attributes.request_type_note.blank')
+      )
+      errors[:request_type_note].any?
+    end
+  end
+
   def update_cached_attributes(new_data_request_log)
     self.cached_date_received = new_data_request_log.date_received
     self.cached_num_pages = new_data_request_log.num_pages
   end
 
   def clean_attributes
-    [:location, :data]
+    [:location]
       .each { |f| self.send("#{f}=", self.send("#{f}")&.strip) }
```

Comment:
maybe add "request_type_note" here instead? although not full sure why here need some clean-up